Makefile

```diff
@@ -13,9 +13,10 @@ UPTODATE := .uptodate
 # $(IMAGE_PREFIX)<dirname>. Dependencies (i.e. things that go in the image)
 # still need to be explicitly declared.
 %/$(UPTODATE): %/Dockerfile %/*
-	$(SUDO) docker build --build-arg=revision=$(GIT_REVISION) -t $(IMAGE_PREFIX)$(shell basename $(@D)) $(@D)/
-	$(SUDO) docker tag $(IMAGE_PREFIX)$(shell basename $(@D)) $(IMAGE_PREFIX)$(shell basename $(@D)):$(IMAGE_TAG)
-	touch $@
+	mkdir -p bin # Restrict the build context to bin, create it here if it doesn't exist.
```

Comment:
This implies that every file which goes into a docker image must live under `bin`.
Although we don't have any examples right now where we want to copy anything besides the binary, I don't think this restriction is reasonable.

Reply:
This is done to accelerate the builds; docker copies the entire build context directory when starting the build which poses a considerable slowdown to CI (assuming CI's filesystem isn't very fast) if we would use the project's root. The previous approach of mixing the Dockerfiles with the go sources also isn't very elegant. Restricting to `bin` was deemed as currently sufficient, if there is a need to build images with additional scripts etc. in the future this can definitely be changed to build e.g. in a temporary directory with the required assets copied/moved in.